packages/client/src/links/internals/httpUtils.ts

```diff
@@ -133,41 +136,79 @@ export const jsonHttpRequester: Requester = (opts) => {
   });
 };
 
+export const streamingJsonHttpRequester: Requester<
+  AsyncGenerator<[index: string, data: HTTPResult], HTTPResult | undefined>
+> = (opts) => {
+  const ac = opts.AbortController ? new opts.AbortController() : null;
+  const responsePromise = getHttpResponse(
+    {
+      ...opts,
+      contentTypeHeader: 'application/json',
+      batchModeHeader: 'stream',
+      getUrl,
+      getBody,
+    },
+    ac,
+  );
+  const cancel = () => ac?.abort();
+  const promise = responsePromise.then(async (res) => {
+    if (!res.body) throw new Error('Received response without body');
+    const meta: HTTPResult['meta'] = { response: res };
+    return parseJsonStream(
+      res.body,
+      (string) => ({
+        json: JSON.parse(string) as TRPCResponse,
+        meta,
+      }),
+      ac?.signal,
+    );
+  });
+
+  return { promise, cancel };
+};
+
 export type HTTPRequestOptions = HTTPBaseRequestOptions &
   ContentOptions & {
     headers: () => HTTPHeaders | Promise<HTTPHeaders>;
   };
 
+async function getHttpResponse(
```

Comment:
I don't think this should be called `getHttpResponse`

Rather like 

```suggestion
async function fetchHTTPResponse(
```

```suggestion
async function performHTTPRequest(
```